src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1382,23 +1706,82 @@ demuxWithDefault_ f kv (Fold dstep dinitial dextract) =
 
     where
 
-    initFold (Fold s i e) = i >>= \r -> return (Fold s (return r) e)
     initial = do
-        mp <- Prelude.mapM initFold kv
-        Tuple' mp <$> dinitial
+        ini <- dinitial
+        mp <- Prelude.mapM initialize kv
+        return $ DemuxingWithDefault ini Set.empty mp
 
-    step (Tuple' mp dacc) a
-      | (k, a') <- f a
-      = case Map.lookup k mp of
-            Nothing -> do
-                acc <- dstep dacc (k, a')
-                return (Tuple' mp acc)
-            Just (Fold step' acc _) -> do
-                _ <- acc >>= \x -> step' x a'
-                return (Tuple' mp dacc)
-    extract (Tuple' mp dacc) = do
+    -- XXX Some code can probably be abstracted
+    step (DemuxingWithDefault dacc bst mp) a =
```

Comment:
This looks ugly.
We should use some `let` statements. Ignore this for now. 
This is changed in `stepify-initial` anyway.